lib/record_store/zone/config.rb

```diff
@@ -3,13 +3,16 @@ class Zone
     class Config
       include ActiveModel::Validations
 
-      attr_reader :ignore_patterns, :provider
+      attr_reader :ignore_patterns, :provider, :supports_alias
+      alias supports_alias? supports_alias
 
       validate :validate_zone_config
 
-      def initialize(ignore_patterns: [], provider: nil)
+      def initialize(ignore_patterns: [], provider: nil, supports_alias: nil)
         @ignore_patterns = ignore_patterns
         @provider = provider
+        @supports_alias = supports_alias
+        @supports_alias = Provider.const_get(provider).supports_alias? if @supports_alias.nil? && valid_provider?
```

Comment:
not 100% happy about putting this here, since it means that Config is no longer as naive as it was, but it gets ugly fast when we try to move it up a level